src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with logical operators (&&, ||) or ternary operator (?:)
-  // without nesting
+  let onlyMen = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    onlyMen = onlyMen.filter(man => Math.ceil(man.died / 100) === century);
+  }
+
+  return onlyMen.reduce(
+    (acc, man, i, arr) => acc + (man.died - man.born) / arr.length, 0);
```

Comment:
if you leave `(` on previous line, `)` - should be on the next line

Reply:
instead of apply filter twice will be better to filter once

Reply:
> instead of apply filter twice will be better to filter once

ok
refactored